apps/omg_eth/lib/eth/root_chain.ex

```diff
@@ -19,82 +19,91 @@ defmodule OMG.Eth.RootChain do
-    contract = contract || Application.get_env(:omg_eth, :contract_addr)
+  def start_exit(utxo_position, txbytes, proof, sigs, from, contract \\ nil, opts \\ nil) do
+    opts = opts || Keyword.put_new(@tx_defaults, :gas, 1_000_000)
+
+    contract = contract || Eth.Encoding.from_hex(Application.get_env(:omg_eth, :contract_addr))
 
     Eth.contract_transact(
       from,
       contract,
       "startExit(uint256,bytes,bytes,bytes)",
       [utxo_position, txbytes, proof, sigs],
-      gasPrice: gas_price,
-      gas: 1_000_000
+      opts
     )
   end
 
-  def deposit(value, from, contract \\ nil) do
-    contract = contract || Application.get_env(:omg_eth, :contract_addr)
-    Eth.contract_transact(from, contract, "deposit()", [], value: value)
+  def deposit(value, from, contract \\ nil, opts \\ @tx_defaults) do
+    opts = Keyword.put(opts, :value, value)
+    contract = contract || Eth.Encoding.from_hex(Application.get_env(:omg_eth, :contract_addr))
+    Eth.contract_transact(from, contract, "deposit()", [], opts)
   end
 
-  def deposit_token(from, token, amount, contract \\ nil) do
-    contract = contract || Application.get_env(:omg_eth, :contract_addr)
+  def deposit_token(from, token, amount, contract \\ nil, opts \\ @tx_defaults) do
+    contract = contract || Eth.Encoding.from_hex(Application.get_env(:omg_eth, :contract_addr))
     signature = "depositFrom(address,address,uint256)"
-    Eth.contract_transact_sync!(from, contract, signature, [Eth.cleanup(from), Eth.cleanup(token), amount])
+    Eth.contract_transact(from, contract, signature, [from, token, amount], opts)
   end
 
-  def add_token(token, contract \\ nil) do
-    contract = contract || Application.get_env(:omg_eth, :contract_addr)
+  def add_token(token, contract \\ nil, opts \\ @tx_defaults) do
+    contract = contract || Eth.Encoding.from_hex(Application.get_env(:omg_eth, :contract_addr))
     {:ok, [from | _]} = Ethereumex.HttpClient.eth_accounts()
-    Eth.contract_transact_sync!(from, contract, "addToken(address)", [token])
+
+    Eth.contract_transact(Eth.Encoding.from_hex(from), contract, "addToken(address)", [token], opts)
   end
 
-  def challenge_exit(cutxopo, eutxoindex, txbytes, proof, sigs, from, contract \\ nil) do
-    contract = contract || Application.get_env(:omg_eth, :contract_addr)
+  def challenge_exit(cutxopo, eutxoindex, txbytes, proof, sigs, from, contract \\ nil, opts \\ @tx_defaults) do
+    contract = contract || Eth.Encoding.from_hex(Application.get_env(:omg_eth, :contract_addr))
     signature = "challengeExit(uint256,uint256,bytes,bytes,bytes)"
     args = [cutxopo, eutxoindex, txbytes, proof, sigs]
-    Eth.contract_transact(from, contract, signature, args)
+    Eth.contract_transact(from, contract, signature, args, opts)
   end
 
-  def create_new(path_project_root, addr) do
+  def create_new(path_project_root, addr, opts \\ nil) do
+    opts = opts || Keyword.put_new(@tx_defaults, :gas, 4_190_937)
+
```

Comment:
why 4_190_937? maybe add constant variable for this number

Reply:
I guess this is a magic number fit for this deployment, but since it's so close to `@lots_of_gas` I'll just use the defaults